src/sentry/static/sentry/app/views/settings/organizationDeveloperSettings/sentryApplicationDetails.jsx

```diff
@@ -0,0 +1,127 @@
+import getDynamicText from 'app/utils/getDynamicText';
+import ApplicationScopes from './applicationScopes';
+
+class SentryApplicationDetails extends AsyncView {
+  static contextTypes = {
+    router: PropTypes.object.isRequired,
+  };
+
+  getDefaultState() {
+    return {
+      ...super.getDefaultState(),
+      app: null,
+    };
+  }
+
+  getEndpoints() {
+    let {appSlug} = this.props.params;
+    if (appSlug) {
+      return [['app', `/sentry-apps/${appSlug}/`]];
+    }
+    return [];
+  }
+
+  getTitle() {
+    return t('Sentry Application Details');
+  }
+
+  handleScopeChange = (onChange, onBlur, scope, scopes, e) => {
+    onChange(scopes, e);
+    onBlur(scopes, e);
+  };
+
+  onSubmitSuccess = data => {
+    const {orgId} = this.props.params;
+    browserHistory.push(`/settings/${orgId}/developer-settings/`);
+  };
+
+  renderBody() {
+    const {orgId} = this.props.params;
+    const {app} = this.state;
+    let method = app ? 'PUT' : 'POST';
+    let endpoint = app ? `/sentry-apps/${app.slug}/` : '/sentry-apps/';
+    return (
+      <div>
+        <SettingsPageHeader title={this.getTitle()} />
+        <Form
+          apiMethod={method}
+          apiEndpoint={endpoint}
+          allowUndo
+          initialData={{organization: orgId, ...app}}
+          onSubmitSuccess={this.onSubmitSuccess}
+          onSubmitError={err => addErrorMessage(t('Unable to save change'))}
+        >
+          <JsonForm location={this.props.location} forms={sentryApplication} />
+          <Panel>
+            <PanelHeader>{t('API Scopes')}</PanelHeader>
+            <PanelBody>
+              <FormField
+                name="scopes"
+                inline={false}
+                flexibleControlStateSize={true}
+                getData={data => ({scopes: data})}
+                required
+              >
+                {({onChange, onBlur}) => (
+                  <ApplicationScopes
+                    onToggle={this.handleScopeChange.bind(this, onChange, onBlur)}
+                    scopes={app && app.scopes ? app.scopes : []}
+                  />
+                )}
+              </FormField>
+            </PanelBody>
+          </Panel>
+          {app && (
+            <Panel>
+              <PanelHeader>{t('Credentials')}</PanelHeader>
+              <PanelBody>
+                <FormField name="clientId" label="Client ID" overflow>
+                  {({value}) => {
+                    return (
+                      <TextCopyInput>
+                        {getDynamicText({value, fixed: 'PERCY_CLIENT_ID'})}
+                      </TextCopyInput>
+                    );
+                  }}
+                </FormField>
+                <FormField
+                  overflow
+                  name="clientSecret"
+                  label="Client Secret"
+                  help={t(`Your secret is only available briefly after application creation. Make
+                      sure to save this value!`)}
```

Comment:
Until we have an easy way to roll these, we'll want to make it available whenever. We can put it behind a "re-confirm your password" thing, though.

Reply:
removing the help text for now - can do the re-confirm password stuff in another PR